### DISCUSSION TITLE
Allow reordering status bar buttons

### DISCUSSION THREAD

**Discussion by uncenter** [zed-industries/zed#26834] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

I would love to have more control over the items in the status bar at the bottom of the editor, beyond simply left or right dock position and enabled/disabled.
I think something like this could work for how it would be configured:

```json
"status_bar_items": {
  "left": ["assistant", "terminal"],
  "right": []
}
```
This would have the assistant and terminal buttons on the left, and nothing on the right.

### Environment

N/A

**Comment by owencmcgrath** [comment]
this would be awesome!

**Comment by marcospb19** [comment]
You can use `"button": false,` to achieve the same results :thinking:.

So this is a suggestion to make the setting a little shorter to type.

In that case, the problem is, you start to get conflicting configs, what if you say it should dock to the right, but the `status_bar_items` says it must dock to the left?

So this issue is non-solvable without breaking a lot of ppls config.

**Comment by uncenter** [comment]
> You can use `"button": false,` to achieve the same results 🤔.

How does that let you specifically reorder the elements on say, the right dock? It doesn't afaik? You can only disable or disable, but not say "go to the left of the xyz button in the right dock".

> In that case, the problem is, you start to get conflicting configs, what if you say it should dock to the right, but the `status_bar_items` says it must dock to the left?
> 
> So this issue is non-solvable without breaking a lot of ppls config.

The setting would be mutually exclusive. Imo ideally the `enabled` and `dock` keys could be phased out/deprecated at some point too.

**Comment by marcospb19** [comment]
> How does that let you specifically reorder the elements on say, the right dock? It doesn't afaik?

Oops, got it.

(I suggest renaming the title to "Allow reordering status bar buttons", to be more specific.)
